derp/derp_server.go

```diff
@@ -157,15 +158,35 @@ type Server struct {
 	// local).
 	clientsMesh map[key.NodePublic]PacketForwarder
 	// sentTo tracks which peers have sent to which other peers,
-	// and at which connection number. This isn't on sclient
-	// because it includes intra-region forwarded packets as the
-	// src.
-	sentTo map[key.NodePublic]map[key.NodePublic]int64 // src => dst => dst's latest sclient.connNum
+	// at which connection number, and how many packets/bytes were
+	// sent per connection. This isn't on sclient because it
+	// includes intra-region forwarded packets as the src.
+	sentTo map[key.NodePublic]map[key.NodePublic]flowData
```

Comment:
I would expect the map key to be a comparable struct value type being a 2-tuple of two NodePublics. Then make the value be a pointer type so you can update it cheaply without doing map operations all the time.

```go
type FlowKey struct {
    Src, Dst key.NodePublic
}

...

    flows map[FlowKey]*flowState
```